cabal-install/Distribution/Client/IndexUtils.hs

```diff
@@ -550,35 +588,45 @@ packageListFromCache mkPkg hnd Cache{..} mode = accum mempty [] mempty cacheEntr
 --
 type BlockNo = Tar.TarEntryOffset
 
-data IndexCacheEntry = CachePackageId PackageId BlockNo
+data IndexCacheEntry = CachePackageId PackageId BlockNo Timestamp
                      | CacheBuildTreeRef BuildTreeRefType BlockNo
-                     | CachePreference Dependency
+                     | CachePreference Dependency BlockNo Timestamp
```

Comment:
What is the BlockNo in the preference case for? We don't need to re-read the preferences file do we?